src/overlays/actors/ovl_En_Mm2/z_en_mm2.c

```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
+    EnMm2* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 21.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06005E18, NULL, this->jointTable, this->morphTable, 16);
+    Animation_Change(&this->skelAnime, sAnimations[2].animation, 1.0f, 0.0f,
+                     Animation_GetLastFrame(sAnimations[2].animation), sAnimations[2].mode, sAnimations[2].morphFrames);
+    this->previousAnimation = 2;
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->unk_1E0 = 0;
+    this->actor.targetMode = 6;
+    this->unk_1F4 |= 1;
+    this->actor.gravity = -1.0f;
+    if (this->actor.params == 1) {
+        this->actionFunc = func_80AAF668;
+    } else {
+        func_80AAEF70(this, globalCtx);
+        this->actionFunc = func_80AAF57C;
+    }
+    if (LINK_IS_CHILD) {
+        Actor_Kill(&this->actor);
+    }
+    if (this->actor.params == 1) {
+        if (!(gSaveContext.infTable[23] & 0x8000) || !(gSaveContext.eventInf[1] & 1)) {
+            osSyncPrintf(VT_FGCOL(CYAN) " マラソン 開始されていない \n" VT_RST "\n");
+            Actor_Kill(&this->actor);
+        }
+    }
+}
+
+void EnMm2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm2* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80AAF224(EnMm2* this, GlobalContext* globalCtx, EnMm2ActionFunc actionFunc) {
+    s16 yawDiff;
+
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->actionFunc = actionFunc;
+        return 1;
+    }
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    if ((ABS(yawDiff) < 0x4301) && (this->actor.xzDistToPlayer < 100.0f)) {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    }
+    return 0;
+}
+
+void func_80AAF2BC(EnMm2* this, GlobalContext* globalCtx) {
+    if (this->unk_1F6 >= 61) {
+        Actor_Kill(&this->actor);
+    }
+    SkelAnime_Update(&this->skelAnime);
+    this->unk_1F6 += 1;
+    Math_SmoothStepToF(&this->actor.speedXZ, 10.0f, 0.6f, 2.0f, 0.0f);
+}
+
+void func_80AAF330(EnMm2* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actionFunc = func_80AAF2BC;
+        EnMm2_ChangeAnimation(this, 0, &this->previousAnimation);
+        this->unk_1E0 = 1;
+        if ((this->unk_1F4 & 2) == 0) {
+            func_80106CCC(globalCtx);
+        }
+        gSaveContext.timer2State = 0;
+        gSaveContext.eventInf[1] &= ~1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/EnMm2_Destroy.s")
+void func_80AAF3C0(EnMm2* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    switch (this->actor.textId) {
+        case 0x607D:
+        case 0x607E: {
```

Comment:
We generally don't use braces in switch cases, so ideally they should be omitted